lib/match.js

```diff
@@ -21,23 +22,29 @@ MatchStream.prototype.scan = function () {
   var self = this
 
   this.opt.redis.eval(lua, 0, this.opt.match, function evalLua (err, res) {
+    handleEvalLua(err, res, self)
+  })
+
+  function handleEvalLua (err, res, self) {
```

Comment:
this should be defined outside of the scope of the caller, so it can be greatly optimized by V8.